Frameworks/UIKit/UITextField.mm

```diff
@@ -322,43 +394,109 @@ - (NSDictionary*)typingAttributes {
+        NSString* passwordString = nil;
+        CGFloat elementWidth = 0.0f;
+
+        // Measure using the password masking character, not the verbatim text
+        if (self->_secureTextMode) {
+            if (_passwordContentElement == nil) {
+                // Not an error, we just might not be loaded yet.
+                [self _applyFont:_adjustedFont];
+                return;
+            }
+            passwordString = [@"" stringByPaddingToLength:self.text.length withString:_passwordBox.passwordChar startingAtIndex:0];
+            elementWidth = _passwordContentElement.actualWidth - _passwordContentElement.padding.left - _passwordContentElement.padding.right;
+        } else {
+            if (_textContentElement == nil) {
+                // Not an error, we just might not be loaded yet.
+                [self _applyFont:_adjustedFont];
+                return;
+            }
+            elementWidth = _textContentElement.actualWidth - _textContentElement.padding.left - _textContentElement.padding.right;
+        }
+
+        // Try smaller and smaller fonts until it fits within the size, or mins out
+        while (_adjustedFont.pointSize > self.minimumFontSize) {
+            // Grab the width directly from the ContentElement, to allow the cancel button to be considered.
+            if (self->_secureTextMode) {
+                CGSize size = [passwordString sizeWithFont:_adjustedFont];
+                if ((_passwordContentElement == nil) || (size.width <= elementWidth)) {
+                    break;
+                }
+            } else {
+                CGSize size = [self.text sizeWithFont:_adjustedFont];
+                if ((_textContentElement == nil) || (size.width <= elementWidth)) {
+                    break;
+                }
+            }
+
+            CGFloat newFontSize = _adjustedFont.pointSize - 1;
+            if (newFontSize < g_minimumFontSize) {
+                _adjustedFont = [_adjustedFont fontWithSize:g_minimumFontSize];
+                break;
+            } else if (newFontSize < self.minimumFontSize) {
+                _adjustedFont = [_adjustedFont fontWithSize:self.minimumFontSize];
+                break;
+            }
+            _adjustedFont = [_adjustedFont fontWithSize:newFontSize];
+        }
+        [self _applyFont:_adjustedFont];
+    } else {
+        [self _applyFont:self.font];
+    }
 }
 
 /**
- @Status Stub
+ @Status Interoperable
 */
 - (void)setMinimumFontSize:(CGFloat)fontSize {
-    UNIMPLEMENTED();
+    _minimumFontSize = fontSize;
+    [self _adjustFontSizeToFitWidthOrApply];
 }
 
 /**
- @Status Stub
+ @Status Interoperable
 */
 - (CGFloat)minimumFontSize {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return _minimumFontSize;
 }
 
 //
 // Managing the Editing Behavior
 //
 /**
- @Status Stub
+ @Status Interoperable
 */
 - (BOOL)isEditing {
-    UNIMPLEMENTED();
-    return StubReturn();
+    if (self.secureTextEntry) {
```

Comment:
Why did we prefer _secureTextMode  over secureTextEntry above? Great if we could be consistent.